vault/cluster/tcp_layer.go

```diff
@@ -64,7 +65,14 @@ func (l *TCPLayer) Listeners() []NetworkListener {
 			l.logger.Info("starting listener", "listener_address", laddr)
 		}
 
-		tcpLn, err := net.ListenTCP("tcp", laddr)
+		// If they've passed 0.0.0.0, we only want to bind on IPv4
+		// rather than golang's dual stack default
+		bindProto := "tcp"
```

Comment:
Does this mean that we don't support IPv6 addrs?

Reply:
We do, but you'd have to provide an IPv6 address here instead if `0.0.0.0`. After thinking about this a bit more i'm going to remove this from this change-set since it's introducing unnecessary risk to the point release.